src/modules/Staff/Service.php

```diff
@@ -84,6 +85,16 @@ public function login($email, $password, $ip)
         }
 
         $this->di['events_manager']->fire(['event' => 'onAfterAdminLogin', 'params' => ['id' => $model->id, 'ip' => $ip]]);
+        if($model -> tfa_token != NULL){
```

Comment:
Does this actually work?
Your code to disable 2FA sets it to an empty string.

Reply:
Yes it works by clearing the secret key in the user table...

When  it is set  and the users provide no token the login fails and the user is asked for the  token.

Then verifies and continue with the login as normal

Reply:
Ah I missed that you wrote that with a loose type comparison, which is the only reason it works.

```PHP
if ('' != null) {
    echo "An empty string is not equal to null when using loose type comparisons";
} else {
    echo "An empty string is equal to null when using loose type comparisons";
}

if ('' !== null) {
    echo "An empty string is not equal to null when using strict type comparisons" ;
} else {
    echo "An empty string is equal to null when using strict type comparisons";
}
```
Gives: 
> An empty string is equal to null when using loose type comparisons
An empty string is not equal to null when using strict type comparisons

I believe our automated code refactoring will change this to be a strict comparison, but even if it doesn't I'd expect us to be using those whenever possible and I think it is incredibly likely that someone would manually change it to be a strict comparison later down the line which will then result in this check not working correctly because your code sets the token in the DB to an [empty string](https://github.com/FOSSBilling/FOSSBilling/pull/2159/files#diff-787f369fd03af5049642284043317f59d420976f86931ccd1d348d8a23325725R57) when resetting it, not to `null`.

Please update the code to reset it back to `null` and then turn this into a strict comparison.

Or arguably even better, skip the comparison check at all and just ensure that it's not a [falsely value](https://www.thisprogrammingthing.com/2021/Truthy-and-Falsy/).
```PHP
if ($model->tfa_token) {
    // Code goes here
}
```